### ISSUE TITLE
[ENH]: Add Universal Color Design categorical palette

### ISSUE THREAD

**Issue by mbhall88** [matplotlib/matplotlib#27757]
### Problem

Hi,

I think the [Color Universal Design](https://jfly.uni-koeln.de/color/) colorblind-friendly palette would be a great addition to matplotlib.

As mentioned in https://github.com/mwaskom/seaborn/issues/3631 this would make sense to add as a categorical colourmap to matplotlib.

### Proposed solution

I have created a repository with some examples of the palette you can take a look at: <https://github.com/mbhall88/cud>

**Comment by ksunden** [comment]
The term "palette" does not actually map to any specific feature in our API.

The closest ideas are:

- The default color cycle
  - This is what is used for standard lines/scatter/patches
  - These are the colors that can be referred to as e.g. `"C0"`
  - We now have a `ColorSequenceRegistry` (at `mpl.color_sequences`)
    - This does little more than provide some names mapped to lists of colors that can be used relatively easily to set the active color cycle.
    - https://matplotlib.org/stable/api/matplotlib_configuration_api.html#matplotlib.color_sequences
  - The active color cycle is one of the things changed by style sheets (which also enable broader style changes)
    - https://matplotlib.org/stable/gallery/style_sheets/style_sheets_reference.html
    
- A qualitative colormap (https://matplotlib.org/stable/users/explain/colors/colormaps.html#qualitative)
  - These _are_ contained in a python level registry by name
  - These are used for `ScalarMappable` data (e.g. images, scatters with a colormap)
  - You _can_ extract the color list from such cmaps to set as color cycles, but it is a little clunky/unusual (internally we use private apis for this)
  
If you are seeking to change the default color cycle, there is extensive conversation on the topic in #9460.

If you would simply like to add one (or more, but keep it relatively small, I'd say) entry into the `mpl.color_sequences` instance by default, I think that would be considered... All current entries into this list are qualitative color maps, so perhaps would do the same... We are generally a _little_ hesitant to add new colormaps, because the more we have the harder it is to know which ones you should use, and it is difficult to change them as they are considered public API.

In general there is a bit of a balancing act here as one of the things that generally makes it easier to differentiate regardless of color perception differences is changes in lightness... however in a color cycle we tend to not want to emphasize any particular values in the cycle and thus having them be the same lightness is desirable, so you only really get to play with hue if you want perceptual uniformity... And then you have to balance having enough colors with having colors that are differentiable. (and different hues have different perceptual lightness with different forms of colorblindness)

**Comment by mwaskom** [comment]
> In general there is a bit of a balancing act here as one of the things that generally makes it easier to differentiate regardless of color perception differences is changes in lightness... however in a color cycle we tend to not want to emphasize any particular values in the cycle and thus having them be the same lightness is desirable

Yeah this is part of why the seaborn `colorblind` palette modifies the UCD values. While the colors are more universally discriminable, the extreme variation in luminance (especially the very dark black color) tends to alter perceptions of the plots even for people with normal color vision — making the dark black elements appear qualitatively quite different, and strongly emphasizing some of the others (bright yellow, vermillion). That bright yellow also has fairly poor constrast against white.

Always  a lot of tradeoffs in data visualization, especially when it come to color!

**Comment by mbhall88** [comment]
> If you are seeking to change the default color cycle, there is extensive conversation on the topic in https://github.com/matplotlib/matplotlib/issues/9460.

Oh no, I certainly am not asking to change the default colour cycle.

You all know a lot more than me so sounds like I should just close this as it maybe doesn't suit being in as a colour map option as there is already a lot and the different hues might not be universal enough.

Happy to just leave it as is and leave this closed issue as a reference for anyone else searching for this colourmap

**Comment by mwaskom** [comment]
IMO this is still worthy of an addition as a qualitative color map in matplotlib!

**Comment by mbhall88** [comment]
Okay, well @ksunden do you think it's worthwhile attempting to add this? What is the process for doing that?

**Comment by mbhall88** [comment]
Is this the place to add it? https://github.com/matplotlib/matplotlib/blob/main/lib/matplotlib/_cm.py

**Comment by QuLogic** [comment]
I did find it a bit weird that the page is on a repository about flies, and not some kind of human research page. They do mention creating a "Color Universal Design Organization" (CUDO), but unfortunately, the link is 404. The site does appear to be working, but is almost entirely in Japanese, so I cannot be 100% sure. We did some looking yesterday, and best we could find was [this page](https://cudo.jp/?page_id=1936), but unfortunately, all the images are dead.

We also found a page on the DIC Global website, which appears to be a pigment/paint/chemical manufacturer. In the section titled [The Color Universal Design–Recommended Color Set](https://www.dic-global.com/en/110th/universal/#sec03), there appears to be a _third edition_ of the colours. At the end of that section, there is a link to [the guidebook in PDF format](https://www.dic-graphics.co.jp/navi/color/ud.html) which appears to contain a _fourth_ edition. There is more guidance in there for 4/5/6 palettes, and also a pastel-looking version.

However, my Japanese is otherwise non-existent, so it would be best to get someone fluent to translate it (the PDF is composed of images, so I cannot copy&paste into translation software.) If we were to add this, it would be best to use the most current edition, but someone does need to untangle what that is (e.g., I don't know if DIC Global is definitive or just their interpretation for their own use.)

**Comment by mbhall88** [comment]
Interesting. I assume you saw this also? https://jfly.uni-koeln.de/color/

**Comment by jklymak** [comment]
I'm still confused if you are trying to make a colormap, or a color sequence.  As noted by @ksunden, we don't have a "palette" concept.  We have colormaps, which are defined in `colors.py` with help from `_cm.py` and `_listed_cm.py`, and stored in a `ColormapRegistry`, accused by `matplotlib.colors.get_cmap`, and is used extensively for color mapping.  

We also have a `ColorSequenceRegistry` that is accessed by `matplotlib.color_sequences`, but I don't see that we have any examples of how you would use that or why.  

Finally, we have style sheets that support a sequence of colors.  I guess that is the eventual plan for `ColorSequenceRegistry`, but it is not currently used that way.  This cycle of colors (`C0`, `C1`...) are probably what you are aiming at.  Note we already have tableau10_colorblind: https://matplotlib.org/stable/gallery/style_sheets/style_sheets_reference.html

**Comment by QuLogic** [comment]
> Interesting. I assume you saw this also? https://jfly.uni-koeln.de/color/ - this is where I got the colourmap for the github repo I linked at the top (https://github.com/mbhall88/cud)

I did; that's the article on the repository about flies. It's also dated 2002/2008, while the others are 2013/2018.

**Comment by mbhall88** [comment]
Okay, clearly my understanding is pretty low here as I don't really understand the difference between a palette and a colourmap etc.

Again, I think I'll just close this and leave it here as a reference for anyone else. Given there is already tableau_10_colorblind that should be more than sufficient
